utils/update-next.js

```
@@ -1,68 +1,160 @@
+      if (pkgJson.ignoreNextjsUpdates) {
+        continue;
+      }
+
+      const dependencies = [pkgJson.dependencies, pkgJson.devDependencies];
+      for (const deps of dependencies) {
+        const oldVersion = deps?.next;
+        if (oldVersion && oldVersion !== newVersion) {
+          updatedCount++;
+          deps.next = newVersion;
+          await writeJson(file, pkgJson, { spaces: 2 });
+
+          // update lock file
+          const cwd = dirname(file);
+          try {
+            if (await pathExists(join(cwd, 'yarn.lock'))) {
+              if (await pathExists(join(cwd, '.yarnrc.yml'))) {
+                console.warn(
+                  `Found ${join(
+                    cwd,
+                    '.yarnrc.yml'
+                  )} and this package probably uses Yarn v2/v3 which is not supported`
+                );
+              } else {
+                exec('yarn', ['generate-lock-entry'], { cwd });
+              }
+            } else if (await pathExists(join(cwd, 'pnpm-lock.yaml'))) {
+              exec('pnpm', ['install', '--lockfile-only'], { cwd });
+            } else if (await pathExists(join(cwd, 'package-lock.json'))) {
+              exec('npm', ['install', '--package-lock-only'], { cwd });
+            }
+          } catch (err) {
+            console.error(`Failed to update next: ${cwd}`);
+            console.error(err.toString());
+          }
+
+          break;
+        }
+      }
+    }
+  }
+
+  if (!updatedCount) {
     console.log(
       `Next.js version ${newVersion} did not change, skipping update.`
     );
     return;
   }
 
-  if (
-    exec('git', ['ls-remote', '--heads', 'origin', branch]).toString().trim()
-  ) {
-    // eslint-disable-next-line no-console
-    console.log(`Branch ${branch} already exists, skipping update.`);
+  console.log(
+    `Updated ${updatedCount} package${
+      updatedCount === 1 ? '' : 's'
+    } to Next.js version ${newVersion}`
+  );
+
+  if (!github || !context) {
```

I think we should throw if these are not defined

---

This is by design so it can be tested locally by running `require('./utils/update-next')()`.